Modules/_decimal/_decimal.c

```diff
@@ -6143,8 +6151,22 @@ decimal_clear(PyObject *module)
     Py_CLEAR(state->SignalTuple);
     Py_CLEAR(state->PyDecimal);
 
-    PyMem_Free(state->signal_map);
-    PyMem_Free(state->cond_map);
+    if (state->signal_map != NULL) {
+        for (DecCondMap *cm = state->signal_map; cm->name != NULL; cm++) {
+            Py_CLEAR(cm->ex);
```

Comment:
You don't need to explicitly set them to `NULL`, since you're freeing the whole map just below. Decref'ing should be enough.